great_expectations/core/expectation_suite.py

```diff
@@ -336,41 +350,116 @@ def find_expectation_indexes(
                 on the data evaluated by that expectation, 'success' to match based on all configuration parameters
                  that influence whether an expectation succeeds based on a given batch of data, and 'runtime' to match
                  based on all configuration parameters
+            ge_cloud_id: Great Expectations Cloud id
 
         Returns: A list of indexes of matching ExpectationConfiguration
 
         Raises:
             InvalidExpectationConfigurationError
 
         """
-        if not isinstance(expectation_configuration, ExpectationConfiguration):
+        if expectation_configuration is None and ge_cloud_id is None:
+            raise TypeError(
+                "Must provide either expectation_configuration or ge_cloud_id"
+            )
+        if expectation_configuration and not isinstance(
+            expectation_configuration, ExpectationConfiguration
+        ):
```

Comment:
We could probably simplify this to `if not isinstance(...)` since an absent config would be `None` and still fail the conditional?